integration/mdc-migration/BUILD.bazel

```diff
@@ -22,13 +67,63 @@ node_integration_test(
         "yarn install --cache-folder .yarn_cache_folder/",
         "yarn ng generate @angular/material:mdc-migration -c all --tsconfig tsconfig.app.json",
         "yarn test",
+        " ".join([
+            "$(rootpath :verify_golden)",
+            "false",
+            "../golden",
+            "integration/mdc-migration/golden",
+        ] + IGNORED_FILES),
+    ],
+    data = [
+        ":golden_project",
+        ":test_project",
+        ":verify_golden",
+    ],
+    npm_packages = npmPackageMappings,
+    setup_chromium = True,
+    tags = [
+        # This test relies on `yarn` so there needs to be internet access.
+        "requires-network",
+    ],
+    # Sample project becomes the working directory for the integration test, ensuring
+    # that its project `package.json` is substitued with the NPM package mappings etc.
+    working_dir = "sample-project/",
+)
+
+# `blaze run` this target to approve new golden.
+node_integration_test(
+    name = "approve",
+    srcs = glob([
+        "golden/**/*",
+        "sample-project/**/*",
+    ]),
+    commands = [
+        # Note: We use a cache folder within the integration test as otherwise
+        # the NPM package mapped archive would be cached in the system.
+        # See: https://github.com/yarnpkg/yarn/issues/2165.
+        # TODO(devversion): determine if a solution/workaround could live in the test runner.
+        "yarn install --cache-folder .yarn_cache_folder/",
+        "yarn ng generate @angular/material:mdc-migration -c all --tsconfig tsconfig.app.json",
+        " ".join([
+            "$(rootpath :verify_golden)",
+            "true",
+            "../golden",
+            "integration/mdc-migration/golden",
+        ] + IGNORED_FILES),
```

Comment:
I think we could avoid this duplication by extracting the target setup to a macro in e.g. `test.bzl` and then have a parameter like the following:

```bzl
migration_test(name = "test", approve = False)
migration_test(name = "test.accept", approve = True)
```

Also note how I follow the pattern `%s.accept` to match with other golden files. This will become important if we at one point support approving all generated files in the repo using a github action. e.g. `/approve-goldens`

Reply:
yeah I didn't put too much work into cleaning it up in this PR. I was thinking in the future we might want to extract it into dev-infra, since running migration golden tests seems like a useful thing for framework too. I'll at least do this to reduce the duplication though